src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -556,9 +559,14 @@ namespace xclcpuemhal2 {
 
     bool isVersal = false;
     std::string binaryDirectory("");
+    
+    //Check if device_process.log already exists. Remove if exists.
+    auto extIoTxtFile = deviceDirectory + "/../../../device_process.log";
```

Comment:
Since we are now using C++17, it would be better if we concatenate directories using std::filesystem::path.  
Doing so will allow for the code to be more "filesystem format" unaware.

Reply:
Pipeline UBUNTU machines 18.04 have gcc version 7.5. We have reverted the std::filesystem usage. As Soren mentioned in Slack, Can we consider this , when all XRT is moving to std::filesystem  at some point when we are ready.

Reply:
That makes sense.